VSDiagnostics/VSDiagnostics/VSDiagnostics/Diagnostics/General/SwitchIsMissingDefaultLabel/SwitchIsMissingDefaultLabelAnalyzer.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using VSDiagnostics.Utilities;
+
+namespace VSDiagnostics.Diagnostics.General.SwitchIsMissingDefaultLabel
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal class SwitchIsMissingDefaultLabelAnalyzer : DiagnosticAnalyzer
+    {
+        private const DiagnosticSeverity Severity = DiagnosticSeverity.Warning;
+
+        private static readonly string Category = VSDiagnosticsResources.GeneralCategory;
+        private static readonly string Message = VSDiagnosticsResources.SwitchIsMissingDefaultSectionAnalyzerMessage;
+        private static readonly string Title = VSDiagnosticsResources.SwitchIsMissingDefaultSectionAnalyzerTitle;
+
+        internal static DiagnosticDescriptor Rule
+            => new DiagnosticDescriptor(DiagnosticId.SwitchIsMissingDefaultLabel, Title, Message, Category, Severity, true);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context) => context.RegisterSyntaxNodeAction(AnalyzeSymbol, SyntaxKind.SwitchStatement);
+
+        private void AnalyzeSymbol(SyntaxNodeAnalysisContext context)
+        {
+            var switchBlock = (SwitchStatementSyntax)context.Node;
```

Comment:
This will cause an exception when you have a parenthesized switch like this:

```csharp
	var x = 5;

	switch ((x))
	{
		case 5: 
		case 6:
		default:
			break;
	}
```

Add this as a test. You can probably just look through the ancestors to find the first of `SyntaxKind.SwitchStatement`.